### PR TITLE
fix(core,platform): table toolbar responsiveness

### LINKED ISSUE #8121: Toolbar - collapsed menu items stay open when closing the overflow
Collapse the overflow example so it's small enough that the "select an option" select is collapsed. Open the overflow menu, then open the collapsed select. Closing the menu will leave the select open and the select body is appended to the top right corner of the window when scrolling.

https://user-images.githubusercontent.com/2471874/169608035-1f8d8df2-d317-4f3f-bea4-5e3c74ce21b7.mov

### LINKED ISSUE #9361: [Reopen][Sourcing][Responsiveness]: fdp table toolbar is not responsive
#### Is this a bug, enhancement, or feature request?
enhancement

#### Briefly describe your proposal.
1) With 200% zoom , If there are many action buttons , then the buttons are overflowing out of container
~~2) With 400% zoom, even if there are only 2 action buttons with search the buttons are overflowing out of container~~
**we only need to account for 200% zoom**

Expected result: The buttons should wrap to next line in the toolbar container

#### Which versions of Angular and Fundamental Library for Angular are affected? (If this is a feature request, use current version.)
angular 13 
core / platform: 0.35.x

#### If this is a bug, please provide steps for reproducing it.
Replicable on https://sap.github.io/fundamental-ngx/#/platform/table 

try adding few more actions buttons say about 4-5 buttons and try zooming to 200%
MicrosoftTeams-image (4)

#### Please provide relevant source code if applicable.
https://sap.github.io/fundamental-ngx/#/platform/table 

#### Is there anything else we should know?
This is crticial for DTAG customer for guided sourcing
priority B

### LINKED ISSUE #9786: Toolbar item groups are not recognized properly
#### Is this a bug, enhancement, or feature request?
Bug
#### Briefly describe your proposal.
Groups are not recognized and all the groups will fall back to `0` group, as a result grouped overflows do not work as advertised
#### Which versions of Angular and Fundamental Library for Angular are affected? (If this is a feature request, use current version.)
Latest and probably since inception of the feature

### LINKED ISSUE #9787: Toolbar does not react to toolbar width change
#### Is this a bug, enhancement, or feature request?
Bug

#### Briefly describe your proposal.
Toolbar overflow does react to only the window resize, but if for some reason toolbar itself is resized, it does not react to that size change and as a result it does not update the items in it

#### Which versions of Angular and Fundamental Library for Angular are affected? (If this is a feature request, use current version.)
latest and from the feature inception quite possibly

### PR DESCRIPTION
**Pull request by g-cheishvili**
## Related Issue(s)

closes #9787
closes #9786
closes #9361
+ performance improvements across the board

## Description
I hope this fixes a lot of the issues

### CODE REVIEW

**Comment by InnaAtanasova** [comment]
1. Toolbar component doc:
Missing space between the items in the overflow menu
Screenshot 2023-05-05 at 10 14 08 AM

2. Platform Table example:
Screenshot 2023-05-05 at 10 15 01 AM

Screenshot 2023-05-05 at 10 15 22 AM

Screenshot 2023-05-05 at 10 15 45 AM
Screenshot 2023-05-05 at 10 15 34 AM

File: libs/cdk/src/lib/utils/components/dynamic-portal/dynamic-portal.component.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+} from '@angular/core';
+import { ComponentPortal, ComponentType, DomPortal, Portal, PortalModule, TemplatePortal } from '@angular/cdk/portal';
+import { coerceElement } from '@angular/cdk/coercion';
+
+@Component({
+    selector: 'fdk-dynamic-portal',
+    standalone: true,
+    imports: [PortalModule],
+    template: ` <ng-template [cdkPortalOutlet]="portal"></ng-template>`
+})
```
Comment (mikerodonnell89) [inline libs/cdk/src/lib/utils/components/dynamic-portal/dynamic-portal.component.ts:19]: Could we get a comment here explaining the purpose of this component? It's unclear just looking at the file

**Comment by g-cheishvili** [comment]
> 1. Toolbar component doc:
>    Missing space between the items in the overflow menu
> 
> Screenshot 2023-05-05 at 10 14 08 AM
> 
> 2. Platform Table example:
> 
> Screenshot 2023-05-05 at 10 15 01 AM
> 
> Screenshot 2023-05-05 at 10 15 22 AM
> 
> Screenshot 2023-05-05 at 10 15 45 AM
> 
> Screenshot 2023-05-05 at 10 15 34 AM

Inna, only the first example is showcasing responsive toolbars, others are left as they were

**Comment by mikerodonnell89** [comment]
> Inna, only the first example is showcasing responsive toolbars, others are left as they were

`shouldOverflow` to default to true in my opinion and user should have to manually set it to false

**Review by InnaAtanasova — APPROVED** [review]

**Review by mikerodonnell89 — APPROVED** [review]

**Pull request merged**